lib/logstash/inputs/syslog.rb

```diff
@@ -145,7 +145,11 @@ def udp_listener(output_queue)
     @logger.info("Starting syslog udp listener", :address => "#{@host}:#{@port}")
 
     @udp.close if @udp
-    @udp = UDPSocket.new(Socket::AF_INET)
+    if @host.include? ':'
+      @udp = UDPSocket.new(Socket::AF_INET6)
+    else
+      @udp = UDPSocket.new(Socket::AF_INET)
+    end
```

Comment:
Wouldn't it be nicer to put:

if IPAddr.new(@host).ipv6?
      @udp = UDPSocket.new(Socket::AF_INET6)
    elsif IPAddr.new(@host).ipv4?
      @udp = UDPSocket.new(Socket::AF_INET)
    end

Reply:
> Wouldn't it be nicer to put:

Feel free to open a new PR to cover IPv6, I believe your approach is much better than mine.

I know (almost?) nothing about Ruby, the PR here is just for reference.